jupyter_notebooks/k_nearest_centroid.ipynb

```diff
@@ -1,16 +1,475 @@
+    "params = ('euclidean', 'cosine', 'manhattan', 'jaccard')\n",
+    "\n",
+    "for metric in params:\n",
+    "    nearest_centroid = NearestCentroid(metric=metric)\n",
+    "    nearest_centroid.fit(data_train, target_train.values.ravel())\n",
+    "    diabetes_test_prediction = nearest_centroid.predict(data_validation)\n",
+    "    print(evaluation(target_validation, diabetes_test_prediction, metric, preprocessing))"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# One Hot Encoding (2) + Undersampling"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "preprocessing = \"One Hot Encoding (2) + Undersampling\"\n",
+    "\n",
+    "#load data\n",
+    "data_train, data_validation, target_train, target_validation = get_preprocessed_brfss_dataset_one_hot_encoded_all_columns_train_test_split_undersampled()\n",
+    "\n",
+    "#metrics that should be used\n",
+    "params = ('euclidean', 'cosine', 'manhattan', 'jaccard')\n",
+    "\n",
+    "for metric in params:\n",
+    "    nearest_centroid = NearestCentroid(metric=metric)\n",
+    "    nearest_centroid.fit(data_train, target_train.values.ravel())\n",
+    "    diabetes_test_prediction = nearest_centroid.predict(data_validation)\n",
+    "    print(evaluation(target_validation, diabetes_test_prediction, metric, preprocessing))"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Evaluation of best approach\n",
+    "We use the following loop to print all F2- and accuracy-scores and thereby also analyze which approach performed best."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "highest_acc = [0.0, None, None]\n",
+    "highest_f2 = [0.0, None, None]\n",
+    "\n",
+    "for i in range(0, len(approach_list)):\n",
+    "    print(\"Nr.{}) {}\".format(i, approach_list[i]))\n",
+    "    print (\"Weighted F2-Score = {}\".format(f2_list[i]))\n",
+    "    print (\"Accuracy-Score = {}\\n\".format(acc_list[i]))\n",
+    "    if highest_f2[0] < float(f2_list[i]):\n",
+    "        highest_f2[0] = f2_list[i]\n",
+    "        highest_f2[1] = i\n",
+    "        highest_f2[2] = approach_list[i]\n",
+    "    if highest_acc[0] < float(acc_list[i]):\n",
+    "        highest_acc[0] = acc_list[i]\n",
+    "        highest_acc[1] = i\n",
+    "        highest_acc[2] = approach_list[i]\n",
+    "\n",
+    "print(\"--------- Best Approaches ---------\")\n",
+    "print(\"Best Approach regarding F2-Score:\\nNr.{}) {} with f2-score = {}\\n\".format(highest_f2[1], highest_f2[2], highest_f2[0]))\n",
+    "print(\"Best Approach regarding Accuracy:\\nNr.{}) {} with acc = {}\".format(highest_acc[1], highest_acc[2], highest_acc[0]))"
```

Comment:
Auch nur als Tipp, in neueren Python Versionen gibt es auch f-strings, welche ich persönlich deutlich schöner finde. Da können Variablen dann direkt in den String geschrieben werden:
`print(f"Best Approach regarding F2-Score: {highest_f2[0]}"`